lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3893,11 +3900,9 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     assert(originalValue->getFunction() == &getOriginal());
     LLVM_DEBUG(getADDebugStream() << "Adding adjoint for " << originalValue);
 #ifndef NDEBUG
-    auto origTy = remapType(originalValue->getType()).getASTType();
-    auto tangentSpace = getTangentSpace(origTy);
     // The adjoint value must be in the tangent space.
-    assert(tangentSpace && newAdjointValue.getType().getASTType()->isEqual(
-               tangentSpace->getCanonicalType()));
+    assert(newAdjointValue.getType() ==
```

Comment:
`getRemappedTangentType` is somewhat expensive, so I kept `#ifndef NDEBUG`.

Reply:
That actually won't make a difference. `assert` is being compiled away anyways.

Reply:
Removing `#ifndef NDEBUG` based on [feedback below](https://github.com/apple/swift/pull/25264#discussion_r290892271).